test/unit/queue/src/QueueTest.cpp

```diff
@@ -82,9 +104,9 @@ TEMPLATE_LIST_TEST_CASE(
     using Fixture = alpaka::test::QueueTestFixture<DevQueue>;
     Fixture f;
 
-    bool CallbackFinished = false;
+    volatile bool CallbackFinished = false;
```

Comment:
I think this is a wrong use of `volatile`. You need a synchronization primitive because `CallbackFinished` is potentially accessed from 2 threads. I would use `std::atomic<bool>`.

Reply:
`volatile` is correct to avoid that the value is cached but I agree that using `std::atomic` makes much more sense because of the access from two threads.

Reply:
You are right in the sense that `volatile` avoids load/store optimizations. This might be correct in this case if there are memory fences somewhere inside `alpaka::enqueue` or `alpaka::wait`. But `std::atomic` is definitely safe, independent on what alpaka or the backends are doing.